miniapps/shifted/marking.cpp

```diff
@@ -103,6 +103,50 @@ void ShiftedFaceMarker::MarkElements(const ParGridFunction &ls_func,
    initial_marking_done = true;
 }
 
+void ShiftedFaceMarker::MarkFaces(mfem::Array<int> &elem_marker,
+                                  mfem::Array<int> &face_marker)
+{
+   pmesh.ExchangeFaceNbrNodes();
+
+   face_marker.SetSize(pmesh.GetNumFaces());
+   face_marker=SBFaceType::UNDEFINED;
+   // Set face_attribute = attr to faces that are on the border between INSIDE and CUT elements.
+   for (int f = 0; f < pmesh.GetNumFaces(); f++)
+   {
+      auto *ft = pmesh.GetFaceElementTransformations(f, 3);
+      if (ft->Elem2No > 0)
+      {
+         if (elem_marker[ft->Elem1No] != elem_marker[ft->Elem2No])
+         {
+            if (elem_marker[ft->Elem1No] == SBElementType::INSIDE)
+            {
+               face_marker[f]=SBFaceType::SURROGATE;
+            }
+            else if (elem_marker[ft->Elem2No] == SBElementType::INSIDE)
+            {
+               face_marker[f]=SBFaceType::SURROGATE;
+            }
+         }
+      }
+   }
+
+   for (int f = 0; f < pmesh.GetNSharedFaces(); f++)
```

Comment:
1. Doesn't pmesh.GetNumFaces() already return the owned and shared faces? If so, why do we need lines 133-147?
2. Can we add some comments to clarify that?